.github/workflows/docker.yml

```diff
@@ -0,0 +1,117 @@
+name: docker release
+
+on:
+  push:
+    branches:
+      - master
+      - stable
+      - docker-release
+    tags:
+
+jobs:
+  pop-runner-instance:
+    name: create and setup infrastructure for branch
+    runs-on: ubuntu-latest
+    env:
+      ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_REGION: ${{ secrets.AWS_REGION }}
+      EC2_SECRET: ${{ secrets.EC2_PWD }}
+      S3_BACKEND_KEY_NAME: tf-state-${{ github.sha }}-docker
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2.1.0
+      - name: setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 0.12.25
+      - name: rewrite tf state key
+        working-directory: ./infrastructure
+        run: sed -i 's/BRANCH_ACTOR_KEY/${{ env.S3_BACKEND_KEY_NAME }}/g' infra.tf
+      - name: Terraform Init
+        working-directory: ./infrastructure
+        run: terraform init
+      - name: Terraform plan
+        working-directory: ./infrastructure
+        run: terraform plan -var="ec2_pwd=$EC2_SECRET" -var="branch_name=${{ github.sha }}-docker"
+      - name: Terraform apply
+        working-directory: ./infrastructure
+        run: terraform apply -var="ec2_pwd=$EC2_SECRET" -var="branch_name=${{ github.sha }}-docker" -auto-approve
+      - name: Sleep for 20 seconds
+        uses: jakejarvis/wait-action@master
+        with:
+          time: '20s'
+
+  docker-build:
+    runs-on: ${{ format('{0}-docker', github.sha) }}
+    needs: pop-runner-instance
+    env:
+      DOCKER_CLI_EXPERIMENTAL: enabled
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Describe version
+        run: git describe --tags --abbrev=0
+      - name: Extract branch name
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+        id: extract_branch
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ secrets.PKG_USER }}
```

Comment:
Should we have this here? Shouldn't we do this in some safer env (like our AMI only after we authenticate this)

Reply:
secret is not accessible to the forks and this workflow is meant to run on push to the master / stable branch